packages/dev/test/NetworkSimultaion.test.ts

```diff
@@ -135,15 +136,7 @@ contract('Network Simulation for Relay Server', function (accounts) {
     describe('repeated boosting', function () {
       const expectedGasPriceAfterSecondBoost = toBN(expectedGasPriceAfterBoost).muln(12).divn(10).toString()
 
-      before('boosting transaction', async function () {
-        await env.relayServer.txStoreManager.clearAll()
-        await env.relayServer.transactionManager._initNonces()
-        await sendMultipleRelayedTransactions()
-        await evmMineMany(pendingTransactionTimeoutBlocks)
-        const latestBlock = await env.web3.eth.getBlock('latest')
-        const allBoostedTransactions = await env.relayServer._boostStuckPendingTransactions(latestBlock.number)
-        assert.equal(allBoostedTransactions.size, stuckTransactionsCount - 1)
-      })
+      before('boosting transaction', assertBoostAndRebroadcast)
```

Comment:
What's going on here? How come you need to extract a function that is both a 'before' clause for this test AND a test on itself within this same describe? This is at least unconventional.

Reply:
That's because when you test something in a waterfall, a test is also a prerequisite state change for another test. It was like that before, I just created a function and added some asserts..